nengo/dists.py

```diff
@@ -99,9 +102,9 @@ class Uniform(Distribution):
         If true, sample from a uniform distribution of integers. In this case,
         low and high should be integers.
     """
-    low = NumberParam()
-    high = NumberParam()
-    integer = BoolParam()
+    low = NumberParam('low')
```

Comment:
Yeah, I don't like it either; there's a bit of discussion about it in the PR comments as I would rather avoid it, though @hunse is OK with it.

To me the main issue is not with `set_param`; for setting these class attributes, I believe that in order to avoid duplication we'd have to do some metaclass stuff since we would essentially be adding some custom stuff to `type.__setattr__` or some other magic method similar to that. Maybe it wouldn't be that bad since it would be part of the base `Distribution` or whatever, but if people try to use `Param`s in their own custom classes it'll be pretty surprising...